### PR TITLE
fix: update naming from era_test_node to anvil-zksync

### PR DESCRIPTION
**Pull request by dutterbutter**
# What :computer: 
* Updates to make use of anvil-zksync

# Why :hand:
* era-test-node renamed to anvil-zksync

# Evidence :camera:
Include screenshots, screen recordings, or `console` output here demonstrating that your changes work as intended

# Notes :memo:
* Any notes/thoughts that the reviewers should know prior to reviewing the code?

### CODE REVIEW

File: packages/hardhat-zksync-node/src/constants.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const MAX_PORT_ATTEMPTS = 10;
 export const PORT_CHECK_DELAY = 500;
 export const RPC_ENDPOINT_PATH = 'eth_chainId';
 
-export const ZKSYNC_ERA_TEST_NODE_NETWORK_NAME = 'zkSyncEraTestNode';
+export const ZKSYNC_ERA_TEST_NODE_NETWORK_NAME = 'anvil-zksync';
```
Comment (Romsters) [inline packages/hardhat-zksync-node/src/constants.ts:41]: Maybe use `AnvilZKsync` for consistency?

File: packages/hardhat-zksync-upgradable/src/core/provider.ts [hunk 2]
```diff
@@ -10,5 +10,5 @@ export const networkNames: { [chainId in number]?: string } = Object.freeze({
     280: 'ZKsync-testnet-goerli',
     270: 'ZKsync-local-setup',
     300: 'ZKsync-testnet-sepolia',
-    260: 'ZKsync-era-test-node',
+    260: 'ZKsync-anvil-zksync',
```
Comment (Romsters) [inline packages/hardhat-zksync-upgradable/src/core/provider.ts:13]: Maybe just `ZKsync-anvil` so zksync doesn't repeat twice?

File: packages/hardhat-zksync/README.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ Here is an overview of the plugins currently available in this package when inst
 | hardhat-zksync-deploy         | Facilitates the deployment of contracts on ZKsync, utilizing artifacts from hardhat-zksync-solc.                            |
 | hardhat-zksync-verify         | Automates the process of verifying smart contracts on the ZKsync network, enhancing transparency and trust.                       |
 | hardhat-zksync-upgradeable    | Enables easier deployment and upgrading of smart contracts on the ZKsync network, improving contract lifecycle management.        |
-| hardhat-zksync-node           | Convenient plugin to run the ZKsync era-test-node locally.                                                                        |
+| hardhat-zksync-node           | Convenient plugin to run the ZKsync anvil-zksync locally.                                                                        |
```
Comment (Romsters) [inline packages/hardhat-zksync/README.md:36]:
```suggestion
| hardhat-zksync-node           | Convenient plugin to run anvil-zksync locally.                                                                        |
```

**Review by Romsters — APPROVED** [review]

**Review by Romsters — CHANGES_REQUESTED** [review]
fix e2e please

**Comment by kiriyaga-txfusion** [comment]
> fix e2e please

Redirecting the era-test-node repository to anvil-zksync will break all older plugins, as the redirection was designed to reduce GitHub API usage and prevent users from hitting API rate limits. This issue is evident in failing e2e tests, which rely on older plugin versions.

**Review by kiriyaga-txfusion — APPROVED** [review]

**Review by Romsters — APPROVED** [review]

**Pull request merged**